### PR TITLE
[SPARK-21121][SQL] Support changing storage level via the spark.sql.inMemoryColumnarStorage.level variable

### PR DESCRIPTION
**Pull request by dosoft**
## What changes were proposed in this pull request?

As described in title

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala [hunk 1]
```diff
@@ -106,6 +106,13 @@ class CacheManager extends Logging {
     }
   }
 
+  def cacheQuery(
+    query: Dataset[_],
+    tableName: Option[String] = None
+  ): Unit = writeLock {
+    cacheQuery(query, tableName, StorageLevel.fromString(query.sparkSession.sessionState.conf.cacheStorageLevel))
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: This line exceeds 100 characters.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala [hunk 2]
```diff
@@ -106,6 +106,13 @@ class CacheManager extends Logging {
     }
   }
 
+  def cacheQuery(
+    query: Dataset[_],
+    tableName: Option[String] = None
+  ): Unit = writeLock {
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: Combine the two lines together

**Comment by gatorsmile** [comment]
Could you add a test case?

**Comment by gatorsmile** [comment]
ok to test

File: sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala [hunk 3]
```diff
@@ -106,6 +105,11 @@ class CacheManager extends Logging {
     }
   }
 
+  def cacheQuery(query: Dataset[_], tableName: Option[String] = None): Unit = writeLock {
+    cacheQuery(query, tableName, StorageLevel.fromString(
+      query.sparkSession.sessionState.conf.cacheStorageLevel))
```
Comment (felixcheung) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: btw, this is a bit odd to roundtrip to string <-> storagelevel

**Comment by HyukjinKwon** [comment]
@dosoft Is this PR active? then would you mind if I ask to reply to the review comment above?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala (hunk 3 above)
Comment (dosoft) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: Could you explain your question in a bit more details?

**Pull request closed without merging**